File: backend/app/utils/access_status.py
from __future__ import annotations

from enum import StrEnum


class AccessStatus(StrEnum):
    OK = "ok"
    BLOCKED = "blocked"
    INACTIVE = "inactive"
    EXPIRED = "expired"
    HABIS = "habis"
    FUP = "fup"


ACCESS_STATUS_ORDER: tuple[str, ...] = (
    AccessStatus.OK,
    AccessStatus.BLOCKED,
    AccessStatus.INACTIVE,
    AccessStatus.EXPIRED,
    AccessStatus.HABIS,
    AccessStatus.FUP,
)


STATUS_PAGE_ALLOWED: set[str] = {
    AccessStatus.BLOCKED,
    AccessStatus.INACTIVE,
    AccessStatus.EXPIRED,
    AccessStatus.HABIS,
    AccessStatus.FUP,
}

File: frontend/types/accessStatus.ts
export const ACCESS_STATUS_VALUES = ['ok', 'blocked', 'inactive', 'expired', 'habis', 'fup'] as const
export type AccessStatus = (typeof ACCESS_STATUS_VALUES)[number]

export const STATUS_PAGE_ALLOWED_VALUES = ['blocked', 'inactive', 'expired', 'habis', 'fup'] as const
